process/factory/shard/vmContainerFactory.go

```diff
@@ -135,20 +160,59 @@ func (vmf *vmContainerFactory) createOutOfProcessArwenVM() (vmcommon.VMExecution
 func (vmf *vmContainerFactory) createInProcessArwenVM() (vmcommon.VMExecutionHandler, error) {
 	logVMContainerFactory.Debug("createInProcessArwenVM", "config", vmf.config)
 
-	return arwenHost.NewArwenVM(
-		vmf.blockChainHookImpl,
-		&arwen.VMHostParameters{
-			VMType:                   factory.ArwenVirtualMachine,
-			BlockGasLimit:            vmf.blockGasLimit,
-			GasSchedule:              vmf.gasSchedule.LatestGasSchedule(),
-			ProtocolBuiltinFunctions: vmf.builtinFunctions,
-			ElrondProtectedKeyPrefix: []byte(core.ElrondProtectedKeyPrefix),
-			ArwenV2EnableEpoch:       vmf.deployEnableEpoch,
-			AheadOfTimeEnableEpoch:   vmf.aheadOfTimeGasUsageEnableEpoch,
-			DynGasLockEnableEpoch:    vmf.deployEnableEpoch,
-			ArwenV3EnableEpoch:       vmf.arwenV3EnableEpoch,
-		},
-	)
+	return vmf.createInProcessArwenVMByEpoch(vmf.initialEpoch)
+}
+
+func (vmf *vmContainerFactory) createInProcessArwenVMByEpoch(epoch uint32) (vmcommon.VMExecutionHandler, error) {
+	if epoch < vmf.ArwenGasManagementRewriteEnableEpoch {
+		return vmf.createInProcessArwenVM_v1_2()
+	}
+
+	return vmf.createInProcessArwenVM_v1_3()
+}
+
+func (vmf *vmContainerFactory) replaceInProcessArwen(epoch uint32) {
+	for _, container := range vmf.containers {
```

Comment:
take out the list of containers. it is not needed. Only ONE container.

Reply:
Done.